omnibus/files/private-chef-cookbooks/private-chef/libraries/ec_postgres.rb

```diff
@@ -3,11 +3,26 @@ class EcPostgres
   def self.with_connection(node, database = 'template1', opts = {})
     require 'pg'
     postgres = node['private_chef']['postgresql'].merge(opts)
-    connection = ::PGconn.open('user' => postgres['db_superuser'],
-                               'host' => postgres['vip'],
-                               'password' => postgres['db_superuser_password'],
-                               'port' => postgres['port'],
-                               'dbname' => database)
+    connection = nil
+    retries = 5
+    begin
+      connection = ::PGconn.open('user' => postgres['db_superuser'],
+                                 'host' => postgres['vip'],
+                                 'password' => postgres['db_superuser_password'],
+                                 'port' => postgres['port'],
+                                 'dbname' => database)
+    rescue e
+      if retries > 0
+        retries -= 1
+        Chef::Log.warn "Error connecting to postgresql: #{e.message}, retrying after 1s sleep. #{retries} retries remaining."
+        sleep 1
```

Comment:
While I'm not sure it'll be neededin the failure cases we're looking at, it can't hurt either. I'll change that.